### ISSUE TITLE
Inserting extraneous slashes

### ISSUE THREAD

**Issue by lyczak** [withfig/fig#570]
Choosing an autocomplete suggestion that includes spaces sometimes inserts backslashes in front of spaces. I have been able to replicate this issue reasonably consistently with the following steps:

1. Open a new zsh session in iTerm
2. Type `git c` -- the first autocomplete result will be `git commit -m 'msg'`
3. Press tab to choose this autocomplete result

This results in the following text being input to the shell prompt:
```
$ git commit\ -m\ 'msg'
```

This problem does not occur when pressing enter instead of tab.
I am using macOS 12.0.1, zsh 5.8, and iTerm 3.4.8 with [powerline-go](https://github.com/justjanne/powerline-go) as my shell prompt and the following zsh plugins (though it's worth noting that I was able to replicate the issue even with zsh-autosuggestions disabled):
```
plugins=(
  git
  zsh-autosuggestions
  zsh-syntax-highlighting
)
```

**Comment by mschrage** [comment]
It is very interesting that the behavior is different between tab and enter. This makes me think that it may be an issue with the autocomplete-engine. cc @sullivan-sean

**Comment by mschrage** [comment]
Hey @lyczak1! This is fixed as of v1.0.52 :)
